cms/toolbar/toolbar.py

```diff
@@ -260,11 +262,19 @@ def get_object_draft_url(self):
 
     # Internal API
 
-    def _add_item(self, item, position):
+    def _add_item(self, item, position=None):
```

Comment:
Because items are placed into their position as all the toolbars are loaded. By separating the lists, it allows the beginning and end to remain independent from one another as all the toolbars get fully loaded.